### PR TITLE
Migrate TestLanguagePicker to react-testing-library

### LINKED ISSUE #11216: Migrate /components/TestLanguagePicker.js to react-testing-library
Migrate /components/TestLanguagePicker.js to react-testing-library

### PR DESCRIPTION
**Pull request by diox**
Fixes #11216

### CODE REVIEW

File: tests/unit/amo/components/TestLanguagePicker.js [hunk 1]
```diff
@@ -1,71 +1,64 @@
...
 
-      expect(root.find('select')).toHaveProp('defaultValue', currentLocale);
+      expect(screen.getByRole('option', { name: 'Français' }).selected).toBe(
+        true,
+      );
+      expect(screen.getByRole('option', { name: 'Français' }).value).toBe('fr');
     });
 
     it('changes the language on change', () => {
-      const _window = { location: '/fr/firefox/' };
```
Comment (diox) [inline tests/unit/amo/components/TestLanguagePicker.js:52]: I never liked this pattern so I spent way too much time to get it to work with the actual `window` object. Is that worth it ? I can't really tell...

File: tests/unit/amo/components/TestLanguagePicker.js [hunk 2]
```diff
@@ -1,71 +1,64 @@
...
+      ).toBeInTheDocument();
     });
 
     it('selects the current locale', () => {
-      const currentLocale = 'fr';
-      const root = renderLanguagePicker({ currentLocale });
+      renderLanguagePicker({ lang: 'fr' });
 
-      expect(root.find('select')).toHaveProp('defaultValue', currentLocale);
+      expect(screen.getByRole('option', { name: 'Français' }).selected).toBe(
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestLanguagePicker.js:46]: Nit, we tend to prefer `toEqual` over `toBe`.

File: tests/unit/amo/components/TestLanguagePicker.js [hunk 3]
```diff
@@ -1,71 +1,64 @@
...
+        true,
+      );
+      expect(screen.getByRole('option', { name: 'Français' }).value).toBe('fr');
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestLanguagePicker.js:49]: As above.

**Review by bobsilverberg — APPROVED** [review]
I have been following a pattern where, when a component is only used in a single parent, then I move the tests for that component into the suite for the parent, and test it in the context of the parent. This makes it into more of an integration test, and also allows us to test the parent, ensuring that it calls the child as expected. With the old method of shallow rendering, we would simply test that the parent called the child passing the expected properties, but we can no longer test that way.

For this component, this would mean that the tests for `LanguagePicker` would live in the test suite for `Footer` (which has not yet been converted), and the tests for `Footer` would live in `TestPage` (which has been converted).

One of the implications for doing it this way is that we are no longer passing properties into the component directly, which means we wouldn't be able to pass `_window`, which is a very roundabout way of saying "yes, it's better to not have to rely on that `_window` property".

We could merge this patch as is, and open a follow-up to move these tests into `TestPage` after `TestFooter` has been converted, or, if you're interested, you could do that work for `TestFooter` as well, and add those to this patch.

r+wc

**Pull request closed without merging**